freeipa/src/main/java/com/sequenceiq/freeipa/converter/backup/BackupConverter.java

```diff
@@ -62,4 +75,28 @@ private void decorateBackupFromLoggingRequest(Backup backup, LoggingRequest logg
             }
         }
     }
+
+    private void decorateBackupFromBackupRequest(Backup backup, BackupRequest backupRequest) {
+        if (backupRequest != null) {
+            backup.setStorageLocation(backupRequest.getStorageLocation());
+            if (backupRequest.getS3() != null) {
+                S3CloudStorageV1Parameters s3Params = new S3CloudStorageV1Parameters();
+                s3Params.setInstanceProfile(backupRequest.getS3().getInstanceProfile());
+                backup.setS3(s3Params);
+            } else if (backupRequest.getAdlsGen2() != null) {
+                AdlsGen2CloudStorageV1Parameters adlsGen2Params = new AdlsGen2CloudStorageV1Parameters();
+                AdlsGen2CloudStorageV1Parameters adlsGen2FromRequest = backupRequest.getAdlsGen2();
+                adlsGen2Params.setAccountKey(adlsGen2FromRequest.getAccountKey());
+                adlsGen2Params.setAccountName(adlsGen2FromRequest.getAccountName());
+                adlsGen2Params.setSecure(adlsGen2FromRequest.isSecure());
+                adlsGen2Params.setManagedIdentity(adlsGen2FromRequest.getManagedIdentity());
+                backup.setAdlsGen2(adlsGen2Params);
+            } else if (backupRequest.getGcs() != null) {
+                GcsCloudStorageV1Parameters gcsParams = new GcsCloudStorageV1Parameters();
+                GcsCloudStorageV1Parameters gcsFromRequest = backupRequest.getGcs();
+                gcsParams.setServiceAccountEmail(gcsFromRequest.getServiceAccountEmail());
+                backup.setGcs(gcsParams);
+            }
```

Comment:
this is not tested

Reply:
Thanks! This is a copy of the original working code where backup location was fetched from the logging request. As I am only separating backup from logging, this will work as exactly before. 

`
private void decorateBackupFromLoggingRequest(Backup backup, LoggingRequest loggingRequest) {
        if (backup != null && loggingRequest != null) {
            backup.setStorageLocation(loggingRequest.getStorageLocation());
            if (loggingRequest.getS3() != null) {
                S3CloudStorageV1Parameters s3Params = new S3CloudStorageV1Parameters();
                s3Params.setInstanceProfile(loggingRequest.getS3().getInstanceProfile());
                backup.setS3(s3Params);
            } else if (loggingRequest.getAdlsGen2() != null) {
                AdlsGen2CloudStorageV1Parameters adlsGen2Params = new AdlsGen2CloudStorageV1Parameters();
                AdlsGen2CloudStorageV1Parameters adlsGen2FromRequest = loggingRequest.getAdlsGen2();
                adlsGen2Params.setAccountKey(adlsGen2FromRequest.getAccountKey());
                adlsGen2Params.setAccountName(adlsGen2FromRequest.getAccountName());
                adlsGen2Params.setSecure(adlsGen2FromRequest.isSecure());
                adlsGen2Params.setManagedIdentity(adlsGen2FromRequest.getManagedIdentity());
                backup.setAdlsGen2(adlsGen2Params);
            } else if (loggingRequest.getGcs() != null) {
                GcsCloudStorageV1Parameters gcsParams = new GcsCloudStorageV1Parameters();
                GcsCloudStorageV1Parameters gcsFromRequest = loggingRequest.getGcs();
                gcsParams.setServiceAccountEmail(gcsFromRequest.getServiceAccountEmail());
                backup.setGcs(gcsParams);
            }
        }
    }
`

Reply:
Thanks! This is a copy of the original working code where backup location was fetched from the logging request. As I am only separating backup from logging, this will work as exactly before.

```
private void decorateBackupFromLoggingRequest(Backup backup, LoggingRequest loggingRequest) {
        if (backup != null && loggingRequest != null) {
            backup.setStorageLocation(loggingRequest.getStorageLocation());
            if (loggingRequest.getS3() != null) {
                S3CloudStorageV1Parameters s3Params = new S3CloudStorageV1Parameters();
                s3Params.setInstanceProfile(loggingRequest.getS3().getInstanceProfile());
                backup.setS3(s3Params);
            } else if (loggingRequest.getAdlsGen2() != null) {
                AdlsGen2CloudStorageV1Parameters adlsGen2Params = new AdlsGen2CloudStorageV1Parameters();
                AdlsGen2CloudStorageV1Parameters adlsGen2FromRequest = loggingRequest.getAdlsGen2();
                adlsGen2Params.setAccountKey(adlsGen2FromRequest.getAccountKey());
                adlsGen2Params.setAccountName(adlsGen2FromRequest.getAccountName());
                adlsGen2Params.setSecure(adlsGen2FromRequest.isSecure());
                adlsGen2Params.setManagedIdentity(adlsGen2FromRequest.getManagedIdentity());
                backup.setAdlsGen2(adlsGen2Params);
            } else if (loggingRequest.getGcs() != null) {
                GcsCloudStorageV1Parameters gcsParams = new GcsCloudStorageV1Parameters();
                GcsCloudStorageV1Parameters gcsFromRequest = loggingRequest.getGcs();
                gcsParams.setServiceAccountEmail(gcsFromRequest.getServiceAccountEmail());
                backup.setGcs(gcsParams);
            }
        }
    }
```